src/main/java/io/supertokens/totp/Totp.java

```diff
@@ -0,0 +1,209 @@
+package io.supertokens.totp;
+
+import io.supertokens.Main;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.totp.TOTPDevice;
+import io.supertokens.pluginInterface.totp.TOTPUsedCode;
+import io.supertokens.pluginInterface.totp.exception.DeviceAlreadyExistsException;
+import io.supertokens.pluginInterface.totp.exception.TotpNotEnabledException;
+import io.supertokens.pluginInterface.totp.exception.UnknownDeviceException;
+import io.supertokens.pluginInterface.totp.sqlStorage.TOTPSQLStorage;
+import io.supertokens.storageLayer.StorageLayer;
+import io.supertokens.totp.exceptions.InvalidTotpException;
+import io.supertokens.totp.exceptions.LimitReachedException;
+
+public class Totp {
+
+    public static String generateSecret() {
+        return "XXXX";
+    }
+
+    public static boolean checkCode(TOTPDevice device, String code) {
+        return true;
+    }
+
+    public static CreateDeviceResponse createDevice(Main main, String userId, String deviceName, int skew, int period)
+            throws StorageQueryException, DeviceAlreadyExistsException {
+
+        TOTPSQLStorage totpStorage = StorageLayer.getTOTPStorage(main);
+        String secret = generateSecret();
+
+        TOTPDevice device = new TOTPDevice(userId, deviceName, secret, skew, period, false);
+        totpStorage.createDevice(device);
+
+        // TODO: Should we just return the secret as a string?
+        return new CreateDeviceResponse(secret);
+    }
+
+    public static VerifyDeviceResponse verifyDevice(Main main, String userId, String deviceName, String code)
```

Comment:
- call verifyCode(main, userId, code, true)
- totpStorage.markDeviceAsVerified(userId, deviceName);
   - if this throws UnknownDeviceException, then catch it -> check if user has any device, and if they don't, then throw TotpNotEnabledException, else throw UnknownDeviceException

Reply:
`checkAndStoreCode()` needs list of devices. And we need to query devices first to find a matching device. 

Why not do a check if the matching device is already verified and return `True` before calling `checkAndStoreCode()`? 

As a result, if the device is already verified, you'll get OK response (with `wasAlreadyVerified=True`) even even if code is invalid. I think is completely normal behaviour. 

Also the frontend SDK, by default, will not call `verifyDevice()` more than once for a device anyways.